docs/source/concepts/interoperability.rst

```diff
@@ -0,0 +1,474 @@
+
+In contrast, `P1` need not output the **Enter** and **Leave** actions at all in this example because `P1` hosts Alice on both ledgers.
+
+.. _cross-ledger-transaction:
+
+Cross-ledger transactions
+=========================
+
+With interoperability, a cross-ledger transaction can be committed on several interoperable DAML ledgers simultaneously.
+Such a cross-ledger transaction avoids some of the synchronization overhead of **Enter** and **Leave** actions.
+When a cross-ledger transaction uses contracts from several DAML ledgers,
+stakeholders may witness actions on their contracts that are actually not visible on the Participant Node.
+
+For example, suppose that the :ref:`paint counteroffer workflow <split-counteroffer-ledger>` from the causality examples is committed as follows:
+The actions on `CounterOffer` and `PaintAgree` contracts are committed on Ledger 1.
+All actions on `Iou`\ s are committed on Ledger 2, assuming that some Participant Node hosts the Bank on Ledger 2.
+The last transaction is a cross-ledger transaction because the archival of the `CounterOffer` and the creation of the `PaintAgree`\ ment commits on Ledger 1 simultaneously with the transfer of Alice's `Iou` to the painter on Ledger 2.
+
+For the last transaction, Participant Node 1 notifies Alice of the transaction tree, the two archivals and the `PaintAgree` creation via the Transaction Service as usual.
+Participant Node 2 also output's the whole transaction tree on Alice's transaction tree stream, which contains the consuming **Exercise** of Alice's `Iou`.
+However, it has not output the **Create** of Alice's `Iou` because `Iou` actions commit on Ledger 2, on which Participant Node 2 does not host Alice.
+So Alice merely *witnesses* the archival even though she is an :ref:`informee <def-informee>` of the exercise.
+The **Exercise** action is therefore marked as merely being witnessed on Participant Node 2's transaction tree stream.
+
+In general, an action is marked as **merely being witnessed** when a party is an informee of the action, but the action is not committed on a ledger on which the Participant Node hosts the party.
+Unlike **Enter** and **Leave**, such witnessed actions do not affect causality from the participant's point of view and therefore provide weaker ordering guarantees.
+Such witnessed actions show up neither in the flat transaction stream nor in the Active Contracts Service.
+
+For example, suppose that the **Create** `PaintAgree` action commits on Ledger 2 instead of Ledger 1, i.e., only the `CounterOffer` actions commit on Ledger 1.
+Then, Participant Node 2 marks the **Create** `PaintAgree` action also as merely being witnessed on the transaction tree stream.
+Accordingly, it does not report the contract as active nor can Alice use the contract in her submissions via Participant Node 2.
+
+.. _interop-causality-graph:
+
+Interoperable causality graphs
```

Comment:
Maybe: "cross-ledger causality graphs", "multi-ledger causality graphs"? Later, "interoperable consistent" sounds a bit off.

Reply:
I'm hesitant about cross-ledger because I already use "cross-ledger transaction" and that's only when a single transaction spans multiple DAML ledgers. But I could live with multi-ledger causality graph and multi-ledger consistent.